src/components/Sections/SectionActions.tsx

```diff
@@ -1,35 +1,122 @@
 import React from 'react'
 
 import ActionsMenuItem from 'cozy-ui/transpiled/react/ActionsMenu/ActionsMenuItem'
-import FileIcon from 'cozy-ui/transpiled/react/Icons/File'
+import Icon from 'cozy-ui/transpiled/react/Icon'
 import ListItemIcon from 'cozy-ui/transpiled/react/ListItemIcon'
 import ListItemText from 'cozy-ui/transpiled/react/ListItemText'
-import { makeActions } from 'cozy-ui/transpiled/react/ActionsMenu/Actions'
-import { Action, DisplayMode } from 'components/Sections/SectionsTypes'
-import Icon from 'cozy-ui/transpiled/react/Icon'
+import ListMin from 'cozy-ui/transpiled/react/Icons/ListMin'
+import MosaicMin from 'cozy-ui/transpiled/react/Icons/MosaicMin'
+import RadioChecked from 'cozy-ui/transpiled/react/Icons/RadioChecked'
+import RadioUnchecked from 'cozy-ui/transpiled/react/Icons/RadioUnchecked'
+import Switch from 'cozy-ui/transpiled/react/Switch'
+import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import {
+  Action,
+  divider,
+  makeActions
+} from 'cozy-ui/transpiled/react/ActionsMenu/Actions'
+import { useSettings } from 'cozy-client'
+
+import {
+  DisplayMode,
+  GroupMode,
+  Section,
+  SectionViewProps
+} from 'components/Sections/SectionsTypes'
+import {
+  computeDisplayMode,
+  handleSectionAction
+} from 'components/Sections/utils'
+
+const isDisplayMode = (
+  value: DisplayMode | GroupMode
+): value is DisplayMode => {
+  return Object.values(DisplayMode).includes(value as DisplayMode)
+}
+
+// This feature is disabled in this pull request
+const StateSwitch = (): JSX.Element => {
+  return <Switch className="u-w-auto u-pr-0" disabled />
+}
 
-const createSectionAction = (primaryText: DisplayMode): (() => Action) => {
-  return () => ({
-    name: `sectionAction_${primaryText}`,
-    action: (_doc, opts): void => {
-      opts.handleAction(primaryText)
-    },
-    Component: React.forwardRef(function SectionActionComponent(props, ref) {
-      return (
-        <ActionsMenuItem {...props} ref={ref}>
-          <ListItemIcon>
-            <Icon icon={FileIcon} />
-          </ListItemIcon>
-          <ListItemText primary={primaryText} />
-        </ActionsMenuItem>
+const createSectionAction = (
+  actionLabel: DisplayMode | GroupMode
+): (() => Action) => {
+  if (isDisplayMode(actionLabel)) {
```

Comment:
I wouldn't have used an if here because I think it makes the function more complex. You could export a function `grouped` which is responsible to create the action. An example, in Photos we make a file for each action ([file](https://github.com/cozy/cozy-photos/blob/f02342a76b217f138f60faaa3907185f017d4a35/src/photos/components/actions/upload.jsx#L11))  so it easy to compose the actions array after. It will help when you will implement the logic to separate it from the rest of the action